opengrok-indexer/src/main/java/org/opengrok/indexer/analysis/Ctags.java

```diff
@@ -390,7 +398,35 @@ public Definitions doCtags(String file) throws IOException,
             if (Thread.interrupted()) {
                 throw new InterruptedException("flush()");
             }
+
+            int timeout = this.timeout * 1000;
+            Timer timer = null;
+            /*
+             * Setup timer thread to make sure the ctags process completes and the indexer can
+             * make progress instead of hanging the whole operation.
+             */
+            if (timeout != 0) {
+                timer = new Timer();
+                timer.schedule(new TimerTask() {
```

Comment:
I was talking about this with Dalibor yesterday and he mentioned that it might produce a cleaner code if you leveraged: https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Future.html#get-long-java.util.concurrent.TimeUnit-

Reply:
thanks @DaliborSkrobak, this indeed looks more Java-y ;-) at the expense of Executor/Exception handling.